src/api/handler.rs

```diff
@@ -1,7 +1,19 @@
+use std::sync::{atomic::AtomicBool, Arc};
+
 use serde_json::json;
 use warp::{reply::json, Rejection, Reply};
 
 // Health handler, responds with { ok: true }
-pub async fn health() -> Result<impl Reply, Rejection> {
-    Ok(json(&json!({"ok": true})))
+pub async fn health(running: Arc<AtomicBool>) -> Result<impl Reply, Rejection> {
+    if running.load(std::sync::atomic::Ordering::Relaxed) {
```

Comment:
Maybe go ahead and import up to `atomic`?

Reply:
I thought I had, might have missed it there 👍